aten/src/ATen/TensorOperators.h

```diff
@@ -37,8 +40,18 @@ inline Tensor& Tensor::operator/=(const Tensor & other) {
 inline Tensor& Tensor::operator/=(Scalar other) {
   return div_(other);
 }
-inline Tensor Tensor::operator[](int64_t idx) const {
-  return select(0, idx);
+inline Tensor Tensor::operator[](Scalar index) const {
+  AT_ASSERT(
+      index.local().isIntegral(),
+      "Can only index tensors with integral scalars (got %s)",
+      index.toTensor().type().toString());
+  return select(0, index.toLong());
+}
+inline Tensor Tensor::operator[](Tensor index) const {
+  // The Scalar(Tensor) constructor is explicit, so we need to actually call
+  // it. This constructor will also throw if the tensor is not zero-dim or
+  // undefined.
```

Comment:
It might be a good idea to raise an error for non-scalar tensors yourself, so users will get a more useful error message

Reply:
Ok